nixos/modules/services/networking/peroxide.nix

```
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.peroxide;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  options.services.peroxide = {
+    enable = mkEnableOption (lib.mdDoc "peroxide service");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.peroxide;
+      defaultText = lib.literalMD "pkgs.peroxide";
+      description = lib.mdDoc "Which peroxide package to use.";
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          UserPortImap = mkOption {
+            type = types.port;
+            default = 1143;
+            description = lib.mdDoc "The port on which to listen for IMAP connections.";
+          };
+
+          UserPortSmtp = mkOption {
+            type = types.port;
+            default = 1025;
+            description = lib.mdDoc "The port on which to listen for SMTP connections.";
+          };
+
+          ServerAddress = mkOption {
+            type = types.str;
+            default = "[::0]";
+            example = "localhost";
+            description = lib.mdDoc "The address on which to listen for connections.";
+          };
+        };
+      };
+      default = { };
+      description = lib.mdDoc ''
+        Configuration for peroxide.  See
+        [config.example.yaml](https://github.com/ljanyst/peroxide/blob/master/config.example.yaml)
+        for an example configuration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.peroxide.settings = {
+      # peroxide deletes the cache directory on startup, which requires write
+      # permission on the parent directory, so we can't use
+      # /var/cache/peroxide
+      CacheDir = "/var/cache/peroxide/cache";
+    };
+
+    users.users.peroxide = {
+      isSystemUser = true;
+      group = "peroxide";
+    };
+    users.groups.peroxide = {};
+
+    systemd.services.peroxide = {
+      requires = [ "network.target" ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = "peroxide";
+        ConfigurationDirectory = "peroxide";
+        ConfigurationDirectoryMode = "0750";
+        LogsDirectory = "peroxide";
```

You're missing [log rotation](https://github.com/ljanyst/peroxide/blob/master/peroxide.logrotate).

---

Upstream [suggestion](https://github.com/ljanyst/peroxide/blob/master/install.sh):
```suggestion
        LogsDirectory = "peroxide";
        LogsDirectoryMode = "0750";
```

---

> You're missing [log rotation](https://github.com/ljanyst/peroxide/blob/master/peroxide.logrotate).

How should that be set in a NixOS module?